mock-identity-certify-plugin/src/test/java/io/mosip/certify/mock/integration/service/MockVCIssuancePluginTest.java

```diff
@@ -0,0 +1,95 @@
+@RunWith(MockitoJUnitRunner.class)
+public class MockVCIssuancePluginTest {
+
+
+    @Mock
+    ParsedAccessToken parsedAccessToken;
+
+    @Mock
+    CacheManager cacheManager;
+
+    @Mock
+    Cache cache=new NoOpCache("test");
+
+    @InjectMocks
+    MockVCIssuancePlugin mockVCIssuancePlugin = new MockVCIssuancePlugin();
+
+    @Mock
+    private SignatureService signatureService;
+
+    private VCRequestDto vcRequestDto = new VCRequestDto();
+
+
+    @Before
+    public void setUp() {
+        ReflectionTestUtils.setField(mockVCIssuancePlugin,"getIdentityUrl","http://example.com");
+        ReflectionTestUtils.setField(mockVCIssuancePlugin,"verificationMethod","http://example.com/verify");
+        ReflectionTestUtils.setField(mockVCIssuancePlugin,"cacheSecretKeyRefId","cacheSecretKeyRefId");
+        ReflectionTestUtils.setField(mockVCIssuancePlugin,"aesECBTransformation","AES/ECB/PKCS5Padding");
+        ReflectionTestUtils.setField(mockVCIssuancePlugin,"storeIndividualId",true);
+        ReflectionTestUtils.setField(mockVCIssuancePlugin,"secureIndividualId",false);
+
+        OIDCTransaction oidcTransaction = new OIDCTransaction();
+        oidcTransaction.setIndividualId("individualId");
+        oidcTransaction.setKycToken("kycToken");
+        oidcTransaction.setAuthTransactionId("authTransactionId");
+        oidcTransaction.setRelyingPartyId("relyingPartyId");
+        oidcTransaction.setClaimsLocales(new String[]{"en-US", "en", "en-CA", "fr-FR", "fr-CA"});
+
+        vcRequestDto.setFormat("ldp_vc");
+        vcRequestDto.setContext(Arrays.asList("context1","context2"));
+        vcRequestDto.setType(Arrays.asList("VerifiableCredential"));
+        vcRequestDto.setCredentialSubject(Map.of("subject1","subject1","subject2","subject2"));
+
+        Mockito.when(cacheManager.getCache(Mockito.anyString())).thenReturn(cache);
+        Mockito.when(parsedAccessToken.getAccessTokenHash()).thenReturn("accessTokenHash");
+    }
+
+    @Test
+    public void getVerifiableCredentialWithLinkedDataProof_withValidDetails_thenPass() throws VCIExchangeException {
+        JWTSignatureResponseDto jwtSignatureResponseDto = new JWTSignatureResponseDto();
+        jwtSignatureResponseDto.setJwtSignedData("test-jwt");
+        Mockito.when(signatureService.jwtSign(any())).thenReturn(jwtSignatureResponseDto);
+        VCResult vcResult = mockVCIssuancePlugin.getVerifiableCredentialWithLinkedDataProof(vcRequestDto,"holderId",Map.of("accessTokenHash","ACCESS_TOKEN_HASH","client_id","CLIENT_ID"));
+        Assert.assertNotNull(vcResult.getCredential());
+        Assert.assertEquals(vcResult.getFormat(),"ldp_vc");
+    }
+
+    @Test
+    public void getVerifiableCredentialWithLinkedDataProof_withInValidIndividualId_thenFail() {
+        try{
+            VCResult result=  mockVCIssuancePlugin.getVerifiableCredentialWithLinkedDataProof(vcRequestDto,"holderId",Map.of("accessTokenHash","ACCESS_TOKEN_HASH","client_id","CLIENT_ID"));
+            Assert.fail();
```

Comment:
I am not able to follow where we are passing the invalid individual id

Reply:
renamed to validIndividualId, expectation is the test will fail if signature service throws error

Reply:
then the test case name is not making sense. Why a valid individual id will fail ?

Reply:
changed the name to getVerifiableCredentialWithLinkedDataProof_withoutSignatureService_thenFail()